### DISCUSSION TITLE
Neotree always visible?

### DISCUSSION THREAD

**Discussion by fandangos** [nvim-neo-tree/neo-tree.nvim#679] [Q&A] [answered]
Hi, first let me thank you for this project. 

I'm new to Neovim and having a file browser makes things much easier. 

I have two questions:

1. How can Neotree opens by default when starting nvim?

I always have to type :Neotree to open it, everytime I start nvim.

3. How can I set it to be present in every tab?

Maybe getting the first question answered will solve this but what I do is, I often open my files with :tabedit since I don't like having split screens to work with. 
But if I open a file using the key T on neotree it will open a new tab but without neotree present on it, so I have to run :Neotree for every tab in nvim.

Hope it's possible as this would make things simpler here.

**Comment by pysan3** [comment, marked as answer]
I think both are achievable with autocmds.

1. How can Neotree opens by default when starting nvim?

First of all, if you launch (a barebone) nvim with a dot `nvim .` at any directory, nvim will open `Netrw`, a built-in file manager, to give a glance on the folder structures.

Neo-tree has a fuctionality of _hijacking_ this behavior to make neo-tree appear instead of netrw which can be configured with `config.filesystem.hijack_netrw_behavior = `[Options see here](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/8238865e1d9c61f1a260c290653f2c419503e0a9/lua/neo-tree/defaults.lua#L450).


If you want to go even further, and make neo-tree appear when just running `nvim`, BUT NOT when `nvim random-file-name.txt`, you can add the following autocmd to somewhere in your config. FYI, `vim.fn.argc()` counts the number of command line arguments. [More Info](https://stackoverflow.com/questions/46216240/vim-open-netrw-on-startup-or-open-files-provided)

```lua
vim.api.nvim_create_autocmd("VimEnter", {
  pattern = "*",
  group = vim.api.nvim_create_augroup("NeotreeOnOpen", { clear = true }),
  once = true,
  callback = function(_)
    if vim.fn.argc() == 0 then
      vim.cmd("Neotree")
    end
  end,
})
```

2. How can I set it to be present in every tab?

There is an autocmd called when a new tab is created. You can simply automate `:Neotree` in that autocmd.

```lua
vim.api.nvim_create_autocmd("TabNew", {
group = vim.api.nvim_create_augroup("NeotreeOnNewTab", { clear = true }),
command = "Neotree",
})
```

**Reply by fandangos** [reply]
Thanks for taking the time to reply to my question. 

Adding the code to my config file actually made it work as I wanted. 
It's much easier to browse files and much faster.

Thank you so much for your help here.

**Reply by AgustinOrdonez** [reply]
Hey, sorry for bothering, I am also new in the vim ecosystem and tried to implement this(Open NeoTree on new tabs), but after adding it I get the following error:
```
Error executing vim.schedule lua callback: ...1/.local/share/nvim/lazy/nui.nvim/lua/nui/utils/init.lua:130: Invalid buffer id: 16
stack traceback:
        [C]: in function 'nvim_buf_set_option'
        ...1/.local/share/nvim/lazy/nui.nvim/lua/nui/utils/init.lua:130: in function 'set_buf_options'
        ...n1/.local/share/nvim/lazy/nui.nvim/lua/nui/tree/init.lua:215: in function 'init'
        .../.local/share/nvim/lazy/nui.nvim/lua/nui/object/init.lua:132: in function 'NuiTree'
        ...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:774: in function 'create_tree'
        ...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1180: in function 'draw'
        ...are/nvim/lazy/neo-tree.nvim/lua/neo-tree/ui/renderer.lua:1319: in function 'show_nodes'
        ...ree.nvim/lua/neo-tree/sources/filesystem/lib/fs_scan.lua:95: in function 'render_context'
        ...ree.nvim/lua/neo-tree/sources/filesystem/lib/fs_scan.lua:158: in function <...ree.nvim/lua/neo-tree/sources/filesystem/lib/fs_scan.lua:157>
```
Am I doing something wrong? I just copy pasted the suggested solution into my init.lua

**Reply by pysan3** [reply]
@AgustinOrdonez Unfortunately I cannot reproduce your issue.

Maybe using autocmd `TabNewEntered` instead of `TabNew` might fix the issue?
Or how about this?

```lua
vim.api.nvim_create_autocmd("TabNewEntered", {
  group = vim.api.nvim_create_augroup("NeotreeOnNewTab", { clear = true }),
  callback = vim.schedule_wrap(function()
    vim.cmd("Neotree show")
  end),
})
```

If this also doesn't fix your issue, please open a [new issue](https://github.com/nvim-neo-tree/neo-tree.nvim/issues/new/choose). I'm very sorry but recently I haven't got spare time, but I'm sure someone in the community can help you.

**Reply by AgustinOrdonez** [reply]
Thanks! The second one worked well. If you still want to reproduce the issue  you could clone [my config](https://github.com/AgustinOrdonez/kickstart.nvim) and add it there (commit was before adding it).
